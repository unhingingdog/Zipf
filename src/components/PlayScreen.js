import React, { Component } from 'react'
import { View, Text, Button } from 'react-native'
import { connect } from 'react-redux'
import { testAction } from '../actions/testAction'
import TextEditor  from './TextEditor'

class PlayScreen extends Component {
	constructor(props) {
		super(props)
		this.pressHandler = this.pressHandler.bind(this)
	}
	
	pressHandler() {
		this.props.testAction()
	}
	
	render() {
		return(<TextEditor test={this.props.test} pressHandler={this.pressHandler} />)
	}
}

const mapStateToProps = state => {
 return	{ 
	 test: state.textReducer.test,
 	}
}

export default connect(mapStateToProps, { testAction })(PlayScreen)