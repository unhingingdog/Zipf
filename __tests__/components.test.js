import PlayScreen from '../src/components/PlayScreen'
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

const middleware = []
const mockStore = configureStore(middleware);

const initialState = {
	textReducer: { test: 'Hello?' }
}

describe('Testing PlayScreen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <PlayScreen />,
      { context: { store: mockStore(initialState) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});



