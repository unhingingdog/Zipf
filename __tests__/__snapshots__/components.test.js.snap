// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PlayScreen renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayScreen
    test="Hello?"
    testAction={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        >
          Hello?
        </Text>,
        <Button
          onPress={[Function]}
          title="some shit"
        />,
      ],
      "style": Object {
        "paddingTop": 50,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Hello?",
          "disabled": false,
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Hello?",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onPress": [Function],
          "title": "some shit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
          >
            Hello?
          </Text>,
          <Button
            onPress={[Function]}
            title="some shit"
          />,
        ],
        "style": Object {
          "paddingTop": 50,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Hello?",
            "disabled": false,
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "Hello?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "some shit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
